discgolfbot/pdga/pdga_cog.py

```diff
@@ -0,0 +1,196 @@
+                await interaction.send(embed=embed, content=f"{interaction.user.mention}:")
+
+    def get_www_pdga_com_user_data(
+            self,
+            pdga_player_number
+    ):
+        try:
+            # construct the www.pdga.com/player/pdga_number scraper and start the scraping
+            pdga_player_scraper = PlayerProfileScraper(pdga_number=f"{pdga_player_number}")
+            pdga_player_scraper.scrape()
+
+            if isinstance(pdga_player_scraper.player_data.player_name, str) and len(pdga_player_scraper.player_data.player_name) >= 3:
+                embed_title = f"www.pdga.com - {pdga_player_scraper.player_data.player_name}"
+            else:
+                embed_title = f"Displaying '{pdga_player_scraper.scrape_url}'"
+                
+            embed = nextcord.Embed(title=embed_title, color=0x004899)
+            if isinstance(pdga_player_scraper.player_data.portrait_url, str) and re.match(pattern="^https{0,1}://", string=pdga_player_scraper.player_data.portrait_url):
+                embed.set_thumbnail(url=pdga_player_scraper.player_data.portrait_url)
+            else:
+                # 1192804555579195492
+                # https://discord.com/channels/1048885791704752158/1048885791704752161/1192806026634862642
+                embed.set_image(url='https://discord.com/assets/ee9c489e574f6ecb1d3c.svg')  # 🥏
+            # call the PlayerDataObject's generate_dict method, creates custom headers 
+            data_dict = pdga_player_scraper.player_data.generate_dict()
+
+            # pass the data to the __process_embed_description_data method
+            desc_contents = self.__process_embed_description_data__(pdga_player_data_dict=data_dict)
+            embed.description = desc_contents
+
+            # filter out the upcoming events, alternatively "N/A" in case there are none # TODO: need to write logic to limit / trim / split upcoming events data if there are too many
+            upcoming_events_data = self.__process_upcoming_events_data__(pdga_player_data_dict=data_dict)
+            embed.add_field(name="Upcoming events:", value=upcoming_events_data, inline=True)
+        except:
+            if not 'embed' in locals():
+                embed_title = "Sorry, I couldn't find the embed I was attempting to work on :("
+                embed = nextcord.Embed(title=embed_title, color=0x004899)
+            else:
+                embed_title = "Something went terribly wrong when I tried to get the pdga.com user data"
+                embed = nextcord.Embed(title=embed_title, color=0x004899)
+        finally:
+            return embed
+        
+    def __process_upcoming_events_data__(self, pdga_player_data_dict:dict):
+            if 'Upcoming Events' in pdga_player_data_dict and isinstance(pdga_player_data_dict, dict):
+                if len(pdga_player_data_dict["Upcoming Events"]) >= 1:
+                    return pdga_player_data_dict['Upcoming Events']
+                else:
```

Comment:
```suggestion
                return "N/A"
```
Trenger ikke en else her siden man allerede returnerer over